ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -151,18 +152,78 @@ private[server] object ServerHelpers {
         }
         .drain
 
-    sg.server[F](bindAddress, additionalSocketOptions = additionalSocketOptions)
+    val handler = sg.server[F](bindAddress, additionalSocketOptions = additionalSocketOptions)
       .interruptWhen(shutdown.signal.attempt)
-      // Divorce the scopes of the server stream and handler streams so the
-      // former can be terminated while handlers complete.
-      .prefetch
       .map { connect =>
         shutdown.trackConnection >>
           Stream
             .resource(connect.flatMap(upgradeSocket(_, tlsInfoOpt)))
             .flatMap(withUpgradedSocket(_))
       }
-      .parJoin(maxConcurrency)
-      .drain
+      
+    forking(handler)
+  }
+
+  // Similar to parJoin with a few semantic differences:
+  // - Inner stream values are not emitted to the new stream
+  // - The outer stream can terminate and finalize early
+  def forking[F[_], O](streams: Stream[F, Stream[F, O]])(implicit F: Concurrent[F]): Stream[F, INothing] = {
```

Comment:
Super similar to https://github.com/typelevel/fs2/blob/main/core/shared/src/main/scala/fs2/Stream.scala except I've ripped scope leasing and queueing out